src/ansys/dpf/post/selection.py

```diff
@@ -0,0 +1,557 @@
+"""Module containing selection classes.
+
+Selection
+---------
+
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List
+
+if TYPE_CHECKING:
+    from ansys.dpf.post.simulation import Simulation
+
+from typing import Union
+
+from ansys.dpf.core import (
+    Field,
+    Scoping,
+    Workflow,
+    locations,
+    operators,
+    time_freq_scoping_factory,
+    types,
+)
+from ansys.dpf.core.common import natures
+from ansys.dpf.core.field import _get_size_of_list
+from ansys.dpf.core.server import get_or_create_server
+from ansys.dpf.core.server_types import BaseServer
+from numpy import ndarray
+
+from ansys.dpf.post.mesh import Mesh
+
+
+class _WfNames:
+    data_sources = "data_sources"
+    scoping = "scoping"
+    scoping_a = "scoping_a"
+    scoping_b = "scoping_b"
+    streams = "streams"
+
+
+class TimeFreqSelection:
+    """Define a time/frequency selection.
+
+    Translate time/frequency selection types (index, sets, time/frequency values) into DPF known
+    entities.
+    """
+
+    def __init__(self, server: Union[BaseServer, None] = None):
+        """Create a time/frequency selection.
+
+        Parameters
+        ----------
+        server:
+            DPF server on which to create the TimeFreqSelection.
+        """
+        self._server = get_or_create_server(server)
+        self._selection = Workflow(server=self._server)
+
+    def select_time_freq_indices(self, time_freq_indices: List[int]) -> None:
+        """Select time frequency sets by their indices (0 based).
+
+        Parameters
+        ----------
+        time_freq_indices:
+            Indices of the time-steps/frequencies to select.
+        """
+        time_freq_sets = [d + 1 for d in time_freq_indices]
+        self.select_time_freq_sets(time_freq_sets)
+
+    def select_time_freq_sets(self, time_freq_sets: List[int]) -> None:
+        """Select time frequency sets by their cumulative sets (1 based).
+
+        Parameters
+        ----------
+        time_freq_sets:
+            Cumulative set indices of the time-steps/frequencies to select.
+        """
+        sets = time_freq_scoping_factory.scoping_by_sets(
+            time_freq_sets, server=self._server
+        )
+        op = operators.utility.forward(sets, server=self._server)
```

Comment:
Because this allows accessing the scoping obtained beforehand using an operator Output to connect it to a further workflow. The selection ends-up returning a workflow, so we need named ouputs and inputs to connect it with another workflow.